lib/algae/maybe.ex

```diff
@@ -54,9 +54,43 @@ defmodule Algae.Maybe do
       iex> new(9)
       %Algae.Maybe.Just{just: 9}
 
+      iex> new(nil)
+      %Algae.Maybe.Just{just: nil}
+
+      iex> new(nil, nothing: nil)
+      %Algae.Maybe.Nothing{}
+
+      iex> new(9, nothing: 9)
+      %Algae.Maybe.Nothing{}
+
+      iex> new(9, nothing: 1)
+      %Algae.Maybe.Just{just: 9}
+
+  """
+  @spec new(any(), keyword()) :: Just.t()
+  def new(value, [nothing: nothing_value]) do
+    if value == nothing_value,
+      do: Nothing.new(),
+    else: Just.new(value)
+  end
```

Comment:
Two very minor nitpicks:

1. The spec is missing a `Nothing`

```elixir
@spec new(any(), [nothing: any()]) :: Just.t() | Nothing.t()
```

(Also may as well add the expected key while we're in there ☝️)

2. Have you considered using pattern matching?

```elixir
def new(nope, [nothing: nope]), do: Nothing.new()
def new(value, _), do: Just.new(value)
```

Reply:
@expede Thanks! I didn't know you could match variables with pattern matching, neat! Spec is fixed as well.